pssh/ssh_client.py

```diff
@@ -100,6 +100,14 @@ def __init__(self, host,
                             else host)
         _user = host_config['user'] if 'user' in host_config else None
         user = user if user else _user
+
+        _port = int(host_config['port']) if 'port' in host_config else 22
+        port = port if port else _port
+
+        # FIXME: we take the first key
+        _pkey = paramiko.RSAKey.from_private_key_file(host_config['identityfile'][0]) if 'identityfile' in host_config else None
```

Comment:
Regarding `FIXME: we take the first key` - paramiko only allows for one key to be provided programmatically in any case so not knowing which key to use is moot - we can only use one.

To support multiple keys a `paramiko.SSHAgent` compatible object like the one under `embedded_server.fake_agent.FakeAgent` could be created, both keys added and that passed on as `agent=<agent object>`. See tests for examples, they use this method to test ssh agent behaviour.

However, user supplying his/her own agent is already supported.

Personally I see user reading his/her own keys and supplying an agent object with them as better suited to a library than having to edit an external config file for another unrelated utility for them. Might be just me :)

Some examples of doing this outside unit test code would be good though, can take that on and provide an `SSHAgent` class in the `pssh` package for that purpose.